### PR TITLE
fix: inability to handle long URLs with params properly when clicked on message

### LINKED ISSUE #2317: [🐛] Url handler not working.
## Issue

App package stream-chat-react-native not handle url properly, when click on link message it gives AccessDenied because of package change url signature and date formate.

web package stream-chat-react properly handle that url and download video file on link.

App video 

https://github.com/GetStream/stream-chat-react-native/assets/129855948/7178637f-81fa-4eb7-be6d-365263e49ade

Web video 

https://github.com/GetStream/stream-chat-react-native/assets/129855948/c5489803-bc9c-4ea4-8c23-ce9c042d55e8

### Steps to reproduce

1. Go to ChannelScreen
2. Click on url message
3. get Access denied error

### Expected behavior

Download file or open file on link click.

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

Fixes #2317 

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: package/src/components/Message/MessageSimple/utils/renderText.tsx [hunk 1]
```diff
@@ -141,8 +141,15 @@ export const renderText = <
       : Linking.canOpenURL(url).then((canOpenUrl) => canOpenUrl && Linking.openURL(url));
   };
 
+  let previousLink: string;
```
Comment (santhoshvai) [inline package/src/components/Message/MessageSimple/utils/renderText.tsx:144]:
```suggestion
  let previousLink: string | undefined;
```

File: package/src/components/Message/MessageSimple/utils/renderText.tsx [hunk 2]
```diff
@@ -141,8 +141,15 @@ export const renderText = <
...
   const linkReact: ReactNodeOutput = (node, output, { ...state }) => {
-    const url = node.target;
+    let url: string;
+    if (state?.withinLink) {
```
Comment (santhoshvai) [inline package/src/components/Message/MessageSimple/utils/renderText.tsx:147]:
```suggestion
    if (state?.withinLink && previousLink) {
```

also add comments for some explanation here please

Comment (khushal87) [reply package/src/components/Message/MessageSimple/utils/renderText.tsx]: Done

File: package/src/components/Message/MessageSimple/utils/renderText.tsx (hunk 1 above)
Comment (khushal87) [reply package/src/components/Message/MessageSimple/utils/renderText.tsx]: Done

**Pull request merged**